pkg/clusteragent/admission/mutate/config/config.go

```diff
@@ -163,19 +171,22 @@ func (w *Webhook) mutate(request *admission.MutateRequest) ([]byte, error) {
 	return common.Mutate(request.Raw, request.Namespace, w.Name(), w.inject, request.DynamicClient)
 }
 
-// inject injects DD_AGENT_HOST and DD_ENTITY_ID into a pod template if needed
+// inject injects the following environment variables into the pod template:
+// - DD_AGENT_HOST: the host IP of the node
+// - DD_ENTITY_ID: the entity ID of the pod
+// - DD_EXTERNAL_ENV: the External Data Environment Variable
 func (w *Webhook) inject(pod *corev1.Pod, _ string, _ dynamic.Interface) (bool, error) {
-	var injectedConfig, injectedEntity bool
+	var injectedConfig, injectedEntity, injectedExternalEnv bool
 
 	if pod == nil {
 		return false, errors.New(metrics.InvalidInput)
-
 	}
 
 	if !autoinstrumentation.ShouldInject(pod, w.wmeta) {
```

Comment:
just confirming, if autoinstrumentation is not enabled for the current pod, should we not inject DD_EXTERNAL_ENV?